zerver/openapi/python_examples.py

```diff
@@ -154,26 +155,38 @@ def get_presence(client: Client) -> None:
     # Get presence information of all the users in an organization.
     result = client.get_realm_presence()
     # {code_example|end}
+    assert result["result"] == "success"
     validate_against_openapi_schema(result, "/realm/presence", "get", "200")
 
 
 @openapi_test_function("/default_streams:post")
 def add_default_stream(client: Client) -> None:
+    result = client.add_subscriptions(
+        streams=[
+            {
+                "name": "test stream",
+                "description": "New stream for testing",
+            },
+        ],
+    )
+    stream_id = client.get_stream_id("test stream")["stream_id"]
+
     # {code_example|start}
     # Add a stream to the set of default streams for new users.
-    stream_id = 10
-
     result = client.add_default_stream(stream_id)
     # {code_example|end}
 
+    assert result["result"] == "success"
```

Comment:
This is a good set of changes given that these asserts are needed.

But it also seems like a pretty bad library design choice that these asserts would be needed. What I think a user would expect when using an API binding library like this, in Python or most other languages, is that if the request failed — if it came back with a 4xx or 5xx response instead of 2xx — then the call should throw, not return a result. Relying on the caller to check for an error seems highly error-prone.

(That doesn't mean we should necessarily invest resources in fixing this anytime soon. I'm not sure how many people use these Python bindings.)

Reply:
It looks like[ 5xx responses do throw](https://github.com/zulip/python-zulip-api/blob/20ccb221193abc283295ce0f69fc78cd2267e800/zulip/zulip/__init__.py#L639-L677), but 4xx responses do not.

There are some 4xx errors checked here and notably `validate_against_openapi_schema` will always return `True` without checking the spec for those cases ...

[CODE LINK](https://github.com/zulip/zulip/blob/1728a2c057166d616fbe677d74e792dc19d65c99/zerver/openapi/openapi.py#L468-L473)
```python
    if status_code.startswith("4"):
        # This return statement should ideally be not here. But since
        # we have not defined 400 responses for various paths this has
        # been added as all 400 have the same schema.  When all 400
        # response have been defined this should be removed.
        return True
```

So, in addition to adding these asserts for "success", it might be good to add a commit to assert for "error" in those 4xx tests. Or add a helper function here that checks/asserts the "result" field in the response, which defaults to checking for "success" and checks for "error" if we're explicitly testing an error response.